crates/proof-impl/l1-batch/src/logic.rs

```diff
@@ -36,12 +37,14 @@ pub fn process_l1_batch_proof(zkvm: &impl ZkVmEnv, btc_blockspace_vk: &[u32; 8])
     let mut rollup_params_commitment = None;
 
     for _ in 0..num_inputs {
-        let blkpo: BlockspaceProofOutput = zkvm.read_verified_borsh(btc_blockspace_vk);
-        let header: Header = deserialize(&blkpo.header_raw).unwrap();
+        let blkpo: BlockScanProofOutput = zkvm.read_verified_borsh(btc_blockspace_vk);
 
-        state.check_and_update_continuity(&header, &get_btc_params());
-        deposits.extend(blkpo.deposits);
-        prev_checkpoint = prev_checkpoint.or(blkpo.prev_checkpoint);
+        for blockscan_result in blkpo.blockscan_results {
```

Comment:
This being a nested loop seems a bit weird to me.  Would it ever make sense for there to be more than one here if we're making the batch proof scan multiple blocks itself?  Or am I misunderstanding how this works now?